Reply/app/src/main/java/com/example/reply/ui/ReplyApp.kt

```diff
@@ -0,0 +1,78 @@
+package com.example.reply.ui
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.PermanentNavigationDrawer
+import androidx.compose.material3.ModalNavigationDrawer
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Modifier
+import kotlinx.coroutines.flow.StateFlow
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ReplyApp(
+    windowSize: WindowSize,
+    devicePosture: StateFlow<DevicePosture>
+) {
+    val foldingDevicePosture = devicePosture.collectAsState().value
+
+    /**
+     * This will help us select type of navigation and content type depending on window size and
+     * fold state of the device.
+     *
+     * In the state of folding device If it's half fold in BookPosture we want to avoid content
+     * at the crease/hinge
+     */
+    val (navigationType, contentType) = when (windowSize) {
+        WindowSize.Compact -> {
+            Pair(ReplyNavigationType.BOTTOM_NAVIGATION, ReplyContentType.LIST_ONLY)
+        }
+        WindowSize.Medium -> {
+            if (foldingDevicePosture != DevicePosture.NormalPosture) {
+                Pair(ReplyNavigationType.NAVIGATION_RAIL, ReplyContentType.LIST_AND_DETAIL)
+            } else {
+                Pair(ReplyNavigationType.NAVIGATION_RAIL, ReplyContentType.LIST_ONLY)
+            }
+        }
+        WindowSize.Expanded -> {
+            if (foldingDevicePosture is DevicePosture.BookPosture) {
+                Pair(ReplyNavigationType.NAVIGATION_RAIL, ReplyContentType.LIST_AND_DETAIL)
+            } else {
+                Pair(ReplyNavigationType.FIXED_NAVIGATION_DRAWER, ReplyContentType.LIST_AND_DETAIL)
+            }
+        }
+    }
```

Comment:
Optional Kotlin trick:

You can separate the declaration from the initialization, and Kotlin will be smart enough to verify that in all cases it was initialized once and exactly once. That might be clearer than the `Pair` approach:

```suggestion
    val navigationType: ReplyNavigationType
    val contentType: ReplyContentType
    when (windowSize) {
        WindowSize.Compact -> {
            replyNavigationType = ReplyNavigationType.BOTTOM_NAVIGATION
            replyContentType = ReplyContentType.LIST_ONLY
        }
        WindowSize.Medium -> {
            replyNavigationType = ReplyNavigationType.NAVIGATION_RAIL
            replyContentType = if (foldingDevicePosture != DevicePosture.NormalPosture) {
                ReplyContentType.LIST_AND_DETAIL
            } else {
                ReplyContentType.LIST_ONLY
            }
        }
        WindowSize.Expanded -> {
            replyNavigationType = if (foldingDevicePosture is DevicePosture.BookPosture) {
               ReplyNavigationType.NAVIGATION_RAIL
            } else {
               ReplyNavigationType.FIXED_NAVIGATION_DRAWER
            }
            replyContentType = ReplyContentType.LIST_AND_DETAIL
        }
    }
```